relayer/chain/ethereum/writer.go

```diff
@@ -62,11 +62,20 @@ func (wr *Writer) Start(ctx context.Context, eg *errgroup.Group) error {
 	return nil
 }
 
+func (wr *Writer) onDone(ctx context.Context) error {
+	wr.log.Info("Shutting down writer...")
+	// Avoid deadlock if a listener is still trying to send to a channel
+	for m := range wr.messages {
+		wr.log.WithFields(logrus.Fields{"message": m}).Debug("Discarded")
```

Comment:
logrus has a shorter api when only passing a single field:
```suggestion
		wr.log.WithField("message", m).Debug("Discarded")
```